steps/xml_steps.py

```diff
@@ -53,7 +53,8 @@ def check_xpath_internal(context, xml_file, xpath, value, strip):
             if result == safe_cast_int(result):
                 if safe_cast_int(result) == safe_cast_int(value):
                     return True
-                raise Exception('Expected element count of %s but got %s' % (value, result))
+                if time.time() >= start_time + TIMEOUT:
```

Comment:
I'm not exactly sure if this is the proper fix. This means that we will wait `TIMEOUT` seconds to actually raise this Exception. If someone decides to set timeout to 10 minutes, we will wait 10 minutes for the test to finish in case of failures.

What was the issue that triggered this fix?

Reply:
I noticed that if I did
```
Feature: EAP Openshift datasources with adjusted base config
@wip
  Scenario: ExampleDS already in config is not changed
    #wildfly-cekit-modules has been updated to only add the default ds if ENABLE_GENERATE_DEFAULT_DATASOURCE=true
    When container is started with command bash
       | variable                           | value           |
       | ENABLE_GENERATE_DEFAULT_DATASOURCE | true            |
    Then copy features/jboss-eap-modules/7/scripts/datasource/add-example-datasource-with-non-std-jndi-name.cli to /tmp in container
    # And run /opt/eap/bin/jboss-cli.sh --file=/tmp/add-example-datasource-with-non-std-jndi-name.cli in container
    And run script -c /opt/eap/bin/openshift-launch.sh /tmp/boot.log in container and detach
    And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should have 1 elements on XPath //*[local-name()='datasource']
    And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should contain value java:jboss/datasources/ExampleDS-original on XPath //*[local-name()='datasource']/@jndi-name
```
It failed (there should be one). After playing with it I found that if I changed the order of a few of the lines (only the relevant shown) to:
```
    And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should contain value java:jboss/datasources/ExampleDS-original on XPath //*[local-name()='datasource']/@jndi-name
    And XML file /opt/eap/standalone/configuration/standalone-openshift.xml should have 1 elements on XPath //*[local-name()='datasource']
```
Now it works, because the 'should contain value' line waits until the config is valid. In the first example, it does not wait at all and fails right away. So although I see your point about the ten minutes, it would have happened for the 'should contain value' check anyway :)

Reply:
Right, this step (`check_xpath_internal`) should not wait for anything by default and only do it if a timeout (new parameter should be added) is specified. Generally steps should return immediately, but this is something for a rewrite of the steps library which will happen at some point. Most probably some helper steps will be added to wait for conditions, but main steps with checks should return immediately.

Reply:
@goldmann the discrepancy is that presently
- the 'should contain value' steps wait until TIMEOUT
- the 'should have x elements' step does not wait at all

If you want, I can try to change this to improve that along what you say

Reply:
Nah, leave it. It needs a bit more thinking and planning, and docs, and...